ddl/ddl.go

```diff
@@ -491,6 +491,11 @@ func (d *ddl) doDDLJob(ctx sessionctx.Context, job *model.Job) error {
 		}
 
 		if historyJob.Error != nil {
+			if historyJob.State == model.JobStateRollbackDone || historyJob.State == model.JobStateCancelled {
+				historyJob.Error = historyJob.Error.Class().New(historyJob.Error.Code(),
```

Comment:
We'd better add a check if `historyJob.Error` is equal to `errCancelledDDLJob`. Then it needs not to return the same error message twice.

Reply:
Thanks, I had modified it.  @zimulala  PTAL